### ISSUE TITLE
ReadFile/WriteFile OVERLAPPED documentation is wrong

### ISSUE THREAD

**Issue by opalmer** [opalmer/pywincffi#86]
From @exvito (https://groups.google.com/d/msg/pywincffi/B9JNULyvhgI/Rbflb9iZBQAJ):

```
PS: By the way, the example code in Read/WriteFile doc strings about
using OVERLAPPED seems odd: AKAICT from the non-trivial MSDN docs, on
Read/WriteFile OVERLAPPED.hEvent should contain a handle to an Event
that will be set when async operations are completed (when working in
async mode). The examples are setting OVERLAPPED.hEvent to the handle
of the file, is this a typo or did I not understand the (admittedly
complex) Read/WriteFile docs?
```

**Comment by exvito** [comment]
Short note - The work I've been doing on the "types" branch has some progress here. It's here https://github.com/exvito/pywincffi/tree/types-cdata-wrapping

**Comment by opalmer** [comment]
I figured, just opened this for tracking purposes mostly.  Didn't want to forget this if you ended up not fixing it in your PR. Once your PR is opened I can comment with "closes #<this issue>" which should close out this issue once your code merges.  

I opened up issues for the other problems you brought up too: #84 #85.  I'll take a look at those myself unless you're planning on adding fixes for those into your code too.
